File: samples/Julia/averages-median.jl
function median2(n)
	s = sort(n)
	len = length(n)
	len%2 == 0 && return (s[ifloor(len/2)+1] + s[ifloor(len/2)])/2
	return s[ifloor(len/2)+1]
end
a = [4.1, 5.6, 7.2, 1.7, 9.3, 4.4, 3.2]
b = [4.1, 7.2, 1.7, 9.3, 4.4, 3.2]
@assert median(a) == median2(a)
@assert median(b) == median2(b)

File: samples/Erlang/averages-median.erl
-module(median).
-import(lists, [nth/2, sort/1]).
-compile(export_all).

median(Unsorted) ->
    Sorted = sort(Unsorted),
    Length = length(Sorted),
    Mid = Length div 2,
    Rem = Length rem 2,
    (nth(Mid+Rem, Sorted) + nth(Mid+1, Sorted)) / 2.
